Prevent only mousedown events to not interfere with touch events

tehapo opened the pull request:
Fixes #288 

Polymer 1.6.0 fixed `preventDefault` on `down` events that were originating from touch events (see https://github.com/Polymer/polymer/pull/3693). Our `vaadin-combo-box` accidentally relied on this bug.

After the update, focusing the input field and scrolling were both broken on touch devices. This PR attempts to resolve the problem by preventing only `mousedown` events.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/vaadin-combo-box/289)
<!-- Reviewable:end -->

tomivirkki commented:
*[vaadin-combo-box-behavior.html, line 160 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/289#-KOn9O-Gg0dndfvEKYMN:-KOn9O-Gg0dndfvEKYMO:b-m5bkya) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/c0e6e1ca583fd5e4cecf48fa43b033bd0491c256/vaadin-combo-box-behavior.html#L160)):*
> ```HTML
>     },
> 
>     _preventMouseDownDefault: function(downEvent) {
> ```

This function with the event type check wouldn't be needed if we removed the invocation from `_onOverlayDown`and added `on-mousedown="_preventDefault"` next to where `_onOverlayDown` is originally bound.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/289)*
<!-- Sent from Reviewable.io -->

tehapo commented:
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[vaadin-combo-box-behavior.html, line 160 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/289#-KOn9O-Gg0dndfvEKYMN:-KOnYMVCdiAWggWriamp:b-2w7u95) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/c0e6e1ca583fd5e4cecf48fa43b033bd0491c256/vaadin-combo-box-behavior.html#L160)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary>
> This function with the event type check wouldn't be needed if we removed the invocation from `_onOverlayDown`and added `on-mousedown="_preventDefault"` next to where `_onOverlayDown` is originally bound.
</details>
Done, definitely a lot cleaner solution now. Thanks!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/289)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 3 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/289#-:-KOo-a9tlrKWdr8pUFvZ:bnfp4nl)*
<!-- Sent from Reviewable.io -->

JesseShaw commented:
I'm using a CDN link in my code.  Is there a temp URL I can test against?

tehapo commented:
@JesseShaw Unfortunately the current CDN setup does not serve the `master` branch versions. However, you could try out [polygit](http://polygit.org/) to import directly from `master`.

See this JS Bin for a working example.
http://jsbin.com/metuselowa/1/edit?html,output
